utils/gyb.py

```diff
@@ -655,7 +659,14 @@ def execute(self, context):
         # Execute the code with our __children__ in scope 
         context.localBindings['__children__'] = self.children
         result = eval(self.code, context.localBindings)
-        assert context.localBindings['__children__'] is self.children
+
+        if context.localBindings['__children__'] is not self.children:
+            raise ValueError(
+                "Expected bindings to be {}, got {}".format(
```

Comment:
This error message isn't precise enough.  It should say, I think, "The code is not allowed to mutate __children__".

Here's a test:
```
%{
__children__ = 42
}%
```